ArcSlider/ArcSlider.js

```diff
@@ -0,0 +1,244 @@
+ *
+ * @module agate/ArcSlider
+ * @exports ArcSlider
+ * @exports ArcSliderBase
+ * @exports ArcSliderDecorator
+ */
+
+import kind from '@enact/core/kind';
+import Pure from '@enact/ui/internal/Pure';
+import ri from '@enact/ui/resolution';
+import Touchable from '@enact/ui/Touchable';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import React from 'react';
+
+import Arc from '../Arc';
+import {angleToPosition} from '../Arc/utils';
+import Skinnable from '../Skinnable';
+
+import ArcSliderBehaviorDecorator from './ArcSliderBehaviorDecorator';
+import {valueToAngle} from './utils';
+
+import css from './ArcSlider.modules.less';
+
+/**
+ * An arc slider component.
+ *
+ * This component is most often not used directly but may be composed within another component as it
+ * is within [ArcSlider]{@link agate/ArcSlider.ArcSlider}.
+ *
+ * @class ArcSliderBase
+ * @memberof agate/ArcSlider
+ * @ui
+ * @public
+ */
+const ArcSliderBase = kind({
+	name: 'ArcSlider',
+
+	propTypes: /** @lends agate/ArcSlider.ArcSliderBase.prototype */ {
+		/**
+		 * The color of the background arc.
+		 *
+		 * @type {String}
+		 * @default #000000
+		 * @public
+		 */
+		backgroundColor: PropTypes.string,
+
+		/**
+		 * Function that generates a reference to the arc svg.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		componentRef: PropTypes.object,
+
+		/**
+		 * The end angle(in degrees) of the arc slider.
+		 *
+		 * The value should be between 0 and 360 and should be greater than startAngle.
+		 *
+		 * @type {String}
+		 * @default 250
+		 * @public
+		 */
+		endAngle: PropTypes.number,
+
+		/**
+		 * The color of the foreground arc.
+		 *
+		 * @type {number}
+		 * @default #0000ff
+		 * @public
+		 */
+		foregroundColor: PropTypes.string,
+
+		/**
+		 * The maximum value of the slider and should be greater than min.
+		 *
+		 * @type {number}
+		 * @default 100
+		 * @public
+		 */
+		max: PropTypes.number,
+
+		/**
+		 * The minimum value of the slider.
+		 *
+		 * @type {Number}
+		 * @default 0
+		 * @public
+		 */
+		min: PropTypes.number,
+
+		/**
+		 * The radius of the arc circle.
+		 *
+		 * @type {Number}
+		 * @default 150
+		 * @public
+		 */
+		radius: PropTypes.oneOf([120, 150]),
+
+		/**
+		 * The start angle(in degrees) of the arc slider.
+		 *
+		 * The value should be between 0 and 360.
+		 *
+		 * @type {number}
+		 * @default 50
+		 * @public
+		 */
+		startAngle: PropTypes.number,
+
+		/**
+		 * The amount to increment or decrement the value.
+		 *
+		 * @type {Number}
+		 * @default 1
+		 * @public
+		 */
+		step: PropTypes.number,
+
+		/**
+		 * The stroke width of the arc slider.
+		 *
+		 * @type {number}
+		 * @default 6
+		 * @public
+		 */
+		strokeWidth: PropTypes.number,
+
+		/**
+		 * The value of the slider.
+		 *
+		 * Defaults to the value of `min`.
+		 *
+		 * @type {Number}
+		 * @public
+		 */
+		value: PropTypes.number
+	},
+
+	defaultProps: {
+		backgroundColor: '#000000',
```

Comment:
For `backgroundColor` and `foregroundColor` default value,
I think these should be retrieved from "skin". WDYT?

Reply:
how can we set a color based on skin on defaultProps?

Reply:
If you refer to `Picker`, you could get from `Skinnable({prop: 'skin'})` as "skin" prop and use this.
I guess we can't use it from defaultProps but from computed.

Reply:
yes, it is possible to get it from computed, but that means the user will not be able to define these colors when using the Arcslider in another component

Reply:
🤔 We could .. but let's just leave this as is.